src/app/auth/services/customer-auth.service.ts

```diff
@@ -0,0 +1,105 @@
+import { HttpClient, HttpErrorResponse } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Observable, catchError, map, tap, throwError } from 'rxjs';
+import { AuthResponse, CustomerResponse } from './interfaces';
+import { SnackbarService } from '../../shared/services/mat-snackbar.service';
+
+export type CustomerRegestrationForm = {
+  email: string;
+  password: string;
+  firstName: string;
+  lastName: string;
+  dateOfBirth: string;
+  addresses: [
+    {
+      streetName: string;
+      city: string;
+      postalCode: string;
+      country: string;
+    },
+  ];
+};
+
+@Injectable({
+  providedIn: 'root',
+})
+export class AuthCustomerService {
+  private apiUrl =
+    'https://api.europe-west1.gcp.commercetools.com/ecommerce-application-rsschool-1905';
```

Comment:
I would move this variable to a separate file and make it a constant to prevent it from being changed.